app/src/main/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentPresenter.kt

```diff
@@ -67,12 +68,11 @@ class TopicLessonsFragmentPresenter @Inject constructor(
     container: ViewGroup?,
     currentExpandedChapterListIndex: Int?,
     expandedChapterListIndexListener: ExpandedChapterListIndexListener,
-    internalProfileId: Int,
     topicId: String,
     storyId: String,
     isDefaultStoryExpanded: Boolean
   ): View? {
-    this.internalProfileId = internalProfileId
+    this.profileId = activity.intent.extractCurrentUserProfileId()
```

Comment:
This should be passed in, instead, since fragments shouldn't be depending on their parent activities directly (as that inverts the dependency chain).

Reply:
Noted, updated the same.